cmd/xl-v1-healing.go

```diff
@@ -667,6 +667,19 @@ func (xl xlObjects) HealObject(ctx context.Context, bucket, object string, dryRu
 	// Read metadata files from all the disks
 	partsMetadata, errs := readAllXLMetadata(healCtx, storageDisks, bucket, object)
 
+	// Check if the object is corrupted, if yes and user requested
+	// remove we simply delete it from namespace, otherwise we
+	if m, ok := xl.isObjectCorrupted(partsMetadata, errs); ok {
+		writeQuorum := m.Erasure.DataBlocks + 1
+		if m.Erasure.DataBlocks == 0 {
+			writeQuorum = len(xl.getDisks())/2 + 1
+		}
+		if remove {
```

Comment:
I think it is better to take dry-run into account here (if !dryRun && remove)

Reply:
Good idea.. @vadmeste